api/main_endpoints/routes/User.js

```diff
@@ -143,12 +143,24 @@ router.post('/users', async function(req, res) {
   };
   const sortOrder = orderToInteger[req.query.order] || orderToInteger.default;
 
-  // make sure that the page we want to see is 0 by default
-  // and avoid negative page numbers
-  let skip = Math.max(Number(req.body.page) || 0, 0);
-  skip *= ROWS_PER_PAGE;
+  // Handle pagination: defaults to page 0 if not specified
+  // Special case: page -1 fetches all users without pagination
+  // All other negative page numbers are clamped to 0
+  let skip, limit;
+  const pageNum = Number(req.body.page);
+
+  if (pageNum === -1) {
+    // Fetch all users (no pagination)
+    skip = 0;
+    limit = 0; // MongoDB uses 0 to mean "no limit"
+  } else {
```

Comment:
is there a situation where we want to fetch every user ever

the pagination approach we have seems fine to me

Reply:
imagine its 3000+ rows to sort on, every time the user changes their input, thats a lot to ask for the browser when the backend is attached to mongodb

can we move this to the backend, and keep the pagination approach, we only ever need to return like the first 25 records